### PR TITLE
Improve fit report

### PR DESCRIPTION
**Pull request by dbrakenhoff**
# Short Description
Add warnings section to fit report including warning 
- if success=False after optimization, 
- if parameters hit bounds
- if response tmax is larger than calibration period

Example:
```
Fit report B40A0611-002||113             Fit Statistics
=======================================================
nfev    580                    EVP                91.49
nobs    233                    R2                  0.91
noise   True                   RMSE                0.15
tmin    2000-01-01 00:00:00    AIC              -944.52
tmax    2014-12-31 00:00:00    BIC              -906.56
freq    D                      Obj                 3.68
warmup  3650 days 00:00:00     ___                     
solver  LmfitSolve             Interp.              Yes

Parameters (11 optimized)
=======================================================
                  optimal    stderr       initial  vary
recharge_A   1.477127e-01   ±22.23%  1.704894e-01  True
recharge_n   8.380223e-01    ±9.67%  8.075562e-01  True
recharge_a   3.062036e+01   ±38.44%  4.279014e+01  True
recharge_f  -5.662943e-01   ±29.78% -5.290465e-01  True
river_A      1.460731e+00   ±19.01%  1.468126e+00  True
river_a      1.000000e+03   ±18.24%  9.999857e+02  True
wells_A     -9.175274e-02   ±40.40% -9.199268e-02  True
wells_a      9.999828e+03   ±92.42%  9.999564e+03  True
wells_b      3.109245e-10  ±137.76%  3.009141e-10  True
constant_d   9.403636e+00    ±9.49%  9.396031e+00  True
noise_alpha  2.192332e+01   ±17.49%  1.400000e+01  True

Warnings! (3)
=======================================================
Parameter 'river_a' on upper bound: 1.00e+03
Response tmax for 'river' > than calibration period
Response tmax for 'wells' > than calibration period
```

# Checklist before PR can be merged:
- [x] closes issue #351
- [x] is documented
- [x] PEP8 compliant code
- [x] tests added / passed
- [ ] ~Example Notebook (for new features)~
- [x] API changes documented in Release Notes

### CODE REVIEW

**Comment by martinvonk** [comment]
Since we are improving the fit report. How about swapping out the EVP for the MAE? :)

**Comment by dbrakenhoff** [comment]
I think that might be quite a change for a lot of our Dutch users... :grimacing:

File: pastas/model.py [hunk 1]
```diff
@@ -1475,17 +1476,18 @@ def _get_file_info(self):
 
         return file_info
 
-    def fit_report(self, output="basic", warnbounds=True):
```
Comment (raoulcollenteur) [inline pastas/model.py:1478]: I think we should add a DeprecationError for warnbounds here, for the off-chance people use it.

e.g., "the 'warnbounds' argument is deprecated, please use 'warnings=True' now.

**Review by raoulcollenteur — CHANGES_REQUESTED** [review]

File: pastas/model.py (hunk 1 above)
Comment (dbrakenhoff) [reply pastas/model.py:1478]: added

**Review by raoulcollenteur — APPROVED** [review]

**Review by raoulcollenteur — APPROVED** [review]

**Pull request merged**